pkg/agent/bakerapi.go

```diff
@@ -29,12 +29,16 @@ func (agentBaker *agentBakerImpl) GetNodeBootstrapping(ctx context.Context,
 		CSE:        templateGenerator.GetNodeBootstrappingCmd(config),
 	}
 
+	distro := config.AgentPoolProfile.Distro
+	if distro == datamodel.CustomizedWindowsOSImage || distro == datamodel.CustomizedImage {
```

Comment:
where is the image coming from in this case? RP sets it from the headers correctly?

Reply:
yes. Before the changes made to use `bakerapi.go`, the behavior was just to use agentbaker to generate CSE and cmd. The logic in vmss.go figured out the right image to use.